source/storing-data/provenance.rst

```diff
@@ -2,3 +2,213 @@ Decentralized Provenance Tracking
+   :alt: A legend indicating how we abbreviate one action's provenance records with single "multiple-files" icon.
+
+With the exception of the current Result (whose provenance lives in ``provenance/``,
+every Action is captured in a directory titled with the Action's :term:`UUID`. 
+That directory contains:
+
+- ``VERSION``: :ref:`identifying-an-archive`
+- ``metadata.yaml``: :ref:`metadata-yaml`
+- ``citations.bib``: all citations related to the run Action, in `bibtex format <https://www.bibtex.com/g/bibtex-format/>`_. (This includes "passthrough" citations like those registered to transformers, regardless of the plugin where they are registered.)
+- ``action/action.yaml``: a YAML description of the Action and its environment. The good stuff!
+- [optional] ``action/metadata.tsv`` or other data files: data captured to provide additional Action context
+
+The ``action.yaml`` file
+````````````````````````
+
+Here, we'll do a deep dive into the contents of a sample visualization's ``action.yaml``.
+These files are broken into three top-level sections, in this order:
+
+- execution: the Action ID and runtime of the Action that created this Result
+- action: Action type, plugin, action, inputs, parameters, etc.
+- environment: a non-comprehensive description of the system and QIIME environment where this action was executed
+
+The specific example shown below is avaiable for your perusal at 
+`q2view <https://view.qiime2.org/provenance/?src=https%3A%2F%2Fdocs.qiime2.org%2F2021.4%2Fdata%2Ftutorials%2Fmoving-pictures%2Fcore-metrics-results%2Funweighted_unifrac_emperor.qzv>`__.
+Click on the bottom square in the provenance graph, 
+or download and open the archive to peruse the YAML file itself.
+
+The execution block
+~~~~~~~~~~~~~~~~~~~
+High-level information about this action and its run time.
+
+.. code-block:: YAML
+
+   execution:
+      uuid: 3611a0c1-e5c5-4308-ac92-ebb5968ebafb
+      runtime:
+          start: 2021-04-21T14:42:16.469998-07:00
+          end: 2021-04-21T14:42:21.080381-07:00
+          duration: 4 seconds, and 610383 microseconds
+
+- Datetimes are formatted as `ISO 8601 timestamps <https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat>`_.
+- The ``uuid`` field captured here is a UUID V4 *representing this Action*, and *not the Result it produced*.
+
+.. admonition:: Maintainer Note
+   :class: maintainer-note
+
+   There are many elements of provenance that require unique IDs
+   (to help us keep track of different aspects of an analysis).
+   Maintaining separate Result and Action IDs
+   (e.g. the ``uuid`` s in ``metadata.yaml`` and ``action.yaml``)
+   allows us to manage the common case where one Action produces multiple Results.
+
+   An added layer of complexity:
+   for Pipelines, the ``uuid`` in the execution block above is actually an alias UUID shared by all Pipeline Actions.
+   The ``alias-of`` UUID shown in the action block below describes the specific Action in that case.
```

Comment:
See my out-of-order note below. Each Pipeline result usually (not always) have a different alias-of pointer.

Reply:
Guess this didn't show up out-of-order after all...